corehq/apps/reports/standard/deployments.py

```diff
@@ -209,8 +210,33 @@ def user_query(self, pagination=True):
                                        )
         return user_query
 
+    def get_location_columns(self, grouped_ancestor_locs):
+        from corehq.apps.locations.models import LocationType
+        location_types = LocationType.objects.by_domain('icds-cas')
+        all_user_locations = grouped_ancestor_locs.values()
+        all_user_loc_types = {loc.location_type_id for user_locs in all_user_locations for loc in user_locs}
+        required_loc_columns = [loc_type for loc_type in location_types if loc_type.id in all_user_loc_types]
+        return required_loc_columns
```

Comment:
nit: these are location types, not columns

Reply:
hey, it was intentional. Because, the requirement is to add the columns for locations.
So to make it general I m using the location type as the column header.
does it sound right to you?. did you gone through the spec I added in the description?

Reply:
Yeah, it makes sense - I was thinking that the method name is confusing, since it sounds like it returns columns rather than location types.